### PR TITLE
chore: Drop Python 3.6 (issue #12)

### PR DESCRIPTION
**Pull request by exoriente**
# Drop Python 3.6 (issue #12 )

Python 3.6 is end-of-life. All references to Python 3.6 have been dropped. Some `if`-statements to handle 3.6 differently in code and dependencies have been removed.

- [x] Issue referenced
- [x] Tests added
- [x] Build OK
- [x] Added changelog entry (section 'next release')

### CODE REVIEW

File: src/quacks/__init__.py [hunk 1]
```diff
@@ -80,14 +78,7 @@ def _is_classvar(t: type) -> bool:
         ...
 
 else:  # pragma: no cover
-    if sys.version_info < (3, 7):
-        from typing import _ClassVar
-
-        def _is_classvar(t: type) -> bool:
-            return type(t) is _ClassVar
+    from typing import ClassVar, _GenericAlias
```
Comment (ariebovenberg) [inline src/quacks/__init__.py:81]: we can remove the TYPE_CHECKING case as well and move the imports to the top, now that there is no separate 3.6 case anymore

File: CONTRIBUTING.rst [hunk 2]
```diff
@@ -18,7 +18,7 @@ Installing the Python version is easy with ``pyenv`` and ``pyenv-virtualenv``.
    brew install pyenv pyenv-virtualenv
 
    # if using pyenv-virtualenv, remember to disable poetry virtualenvs
-   config virtualenvs.create false
+   poetry config virtualenvs.create false
```
Comment (ariebovenberg) [inline CONTRIBUTING.rst:21]: nice catch

**Review by ariebovenberg — CHANGES_REQUESTED** [review]

File: src/quacks/__init__.py (hunk 1 above)
Comment (ariebovenberg) [reply src/quacks/__init__.py:81]: also the coverage exemptions can be removed

Comment (exoriente) [reply src/quacks/__init__.py:81]: I tried, that but `mypy` complains about `_GenericAlias` not being present in `typing.pyi`. I assumed that why it was wrapped in the `TYPE_CHECKING` case.

Comment (ariebovenberg) [reply src/quacks/__init__.py:81]: hmmm 🤔. Does adding `# type: ignore` after that particular import fix it?

Comment (ariebovenberg) [reply src/quacks/__init__.py:81]: hmmm let me see if I can figure this out...

Comment (ariebovenberg) [reply src/quacks/__init__.py:81]: on second thought: let's merge and fix separately

**Comment by ariebovenberg** [comment]
@exoriente let's merge and check later if the type thing can be fixed. 

Could you add a changelog entry first 👀 ?

**Comment by ariebovenberg** [comment]
nvm, I can add the changelog entry easily afterwards. Let's see if I can solve the merge lockfile conflict from my side...

**Review by ariebovenberg — APPROVED** [review]

**Pull request merged**